src/classes/User.js

```diff
@@ -0,0 +1,47 @@
+class User {
+  constructor(user) {
+    this.id = user.id;
+    this.name = user.name;
+    this.bookings = [];
+    this.totalSpent = 0;
+    this.pastBookings = [];
+    this.currentBookings = [];
+  }
+
+  setBookings(bookings) {
+    return this.bookings = bookings.filter(booking => booking.userID === this.id);
+  }
+
+  addBooking(booking) {
+    this.bookings.push(booking);
+  }
+
+  calculateTotalSpent(rooms) {
+    this.totalSpent = this.bookings.reduce((acc, booking) => {
+      let foundRoom = rooms.find(room => room.number === booking.roomNumber)
+      acc += foundRoom.costPerNight;
+      return acc;
+    }, 0);
+    return this.totalSpent.toFixed(2);
+  }
+
+  findCurrentAndPastBookings(date) {
+    const splitTodaysDate = date.split('/');
+
+    this.pastBookings = this.bookings.filter(booking => {
+      const splitBookingDate = booking.date.split('/');
```

Comment:
You could probably combine these conditions into one gigantic condition with || operators, since you are just returning the booking in all of the else blocks.